src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -329,15 +323,32 @@ def _run_task(self, ahs_program):
         return task
 
     def _validate_pulses(self, pulses):
-
-        # ToDo: allow local drive
-
-        if len(pulses) > 1:
-            raise NotImplementedError(
-                f"Multiple pulses in a Rydberg Hamiltonian are not currently supported on "
-                f"hardware. Recieved {len(pulses)} pulses.")
-
-        if pulses[0].wires != self.wires:
-            raise NotImplementedError(
-                f"Only global drive is currently supported on hardware. Found drive defined for subset "
-                f"{[pulses[0].wires]} of all wires [{self.wires}]")
\ No newline at end of file
+        # Iterate through pulses to find global drive
+        max_wires = 0
+        for i, pulse in enumerate(pulses):
+            if len(pulse.wires) > max_wires:
+                max_wires = len(pulse.wires)
+                global_index = i
+
+        local_pulses = pulses.copy()
+        global_pulse = local_pulses.pop(global_index)
+
+        # Validate that global drive covers all wires
+        if global_pulse.wires != self.wires:
+            raise ValueError(
+                "No global drive has been defined. Found drive defined for subset "
+                f"{[global_pulse.wires]} of all wires [{self.wires}]"
+            )
+
+        # Validate that local drives don't have amplitude or phase, and detuning is not time-dependent
+        for pulse in local_pulses:
+            if pulse.amplitude is not None and pulse.amplitude != 0.0:
+                raise ValueError(
+                    "Shifting field only allows specification of detuning. Amplitude must be zero."
+                )
+            if pulse.phase is not None and pulse.phase != 0.0:
+                raise ValueError(
+                    "Shifting field only allows specification of detuning. Phase must be zero."
+                )
+            if callable(pulse.detuning):
```

Comment:
So since callable can be a time-series, we will need to remove this and instead check that the time-series is the same to a scaling factor for all of them, and positive. It might be that we need to add these checks inside the function that creates a ShiftingField instead of having them here, since they would be easiest to do if we had the full time-series.

Reply:
I guess here we could still confirm that they are either all callables or all constants, and do the more complex comparison/validation once we get to a function creating a `ShiftingField`

Reply:
I added this check before we confirmed whether local detuning would be constant or not. I'll update it accordingly now. Since I need to validate the shape of the time series, I think it would make sense to also store and return the calculated pattern in this function.